src/wpt/harness.ts

```diff
@@ -480,22 +739,34 @@ async function validate(
   }
 }
 
-export function run(config: TestRunnerConfig, file: string): TestCase {
-  const options = config[file] ?? {};
-
-  return {
-    async test(_: unknown, env: Env): Promise<void> {
-      if (options.skipAllTests) {
-        console.warn(`All tests in ${file} have been skipped.`);
-        return;
-      }
-
-      prepare(env, options);
-      if (typeof env[file] !== 'string') {
-        throw new Error(`Unable to run ${file}. Code is not a string`);
-      }
-      env.unsafe.eval(env[file]);
-      await validate(file, options);
-    },
+export function createRunner(
+  config: TestRunnerConfig
+): (file: string) => TestCase {
+  return (file: string): TestCase => {
+    return {
+      async test(_: unknown, env: Env): Promise<void> {
+        const options = config[file];
+        if (!options) {
+          throw new Error(
+            `Missing test configuration for ${file}. Specify '${file}': {} for default options.`
+          );
+        }
```

Comment:
@anonrig Here's the bit that requires us to list every test.

Reply:
Thanks!